File: Practice-Solutions/Balloons.java
import java.util.*;

public class Balloons {
    public static void main(String [] args)
    {
        Scanner in = new Scanner(System.in);

        // System.out.print("\nSince this is new, Please enter No. of test cases( 10, ;) ): ");
        int T = in.nextInt();

        // initialize cost of balloons with colors green and purple.
        // System.out.print("Enter cost of balloons (x2): ");
        // int green = in.nextInt();
        // int purple = in.nextInt();
       
        // Do [T] number of times
        for (int testCase = 0; testCase < T; testCase++){
            int green = in.nextInt();
            int purple = in.nextInt();

            // initialize number of participants
            // System.out.print("Enter No. of participants: ");
            int students = in.nextInt();            

            // initialize arrays for holding participants data
            int [] q1 = new int[students];
            int [] q2 = new int[students]; 

            // obtain data for each participant
            // System.out.println();
            for (int i = 0; i < students; i++){
                // System.out.print("Case #"+ i +" enter participants result for the problem (0 or 1): ");
                q1[i] = in.nextInt();
                q2[i] = in.nextInt();
            }

            // System.out.println("\nMyLists -->\nq1: "+ Arrays.toString(q1) +"\nq2: " + Arrays.toString(q2) +"\n<--\n");
            int q1Count = 0, q2Count = 0;

            // iterate over the arrays (q1 & q2) and compare their elements
            for (int sf = 0; sf < students; sf++){
                if (q1[sf] == 1) q1Count++;
                if (q2[sf] == 1) q2Count++;
            }

            // Calculate the sum and determine the minimum cost
            int q1Sum = q1Count * green + q2Count * purple;
            int q2Sum = q1Count * purple + q2Count * green;

            if (q1Sum <= q2Sum){ 
                System.out.println(q1Sum);

            }else{
                System.out.println(q2Sum);
            }
        }
        in.close();

        // System.out.println("T: "+ T);
    }
}

File: Practice-Solutions/Balloons.cpp
#include <bits/stdc++.h>

using namespace std;

int main () {

    // optimize for faster reading of input
    // comment this code if using 'cout' 
    // when reading in input
	ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    // cout <<"\nEnter No. of test cases: ";
    int T;
    cin >> T;

    // Run T test cases
    for (int t = 0; t < T; t++) {
        
        // get cost of green and purple balloons
        // cout <<"\nEnter cost of green and purple balloons: ";
        int costGreen, costPurple, p;
        cin >> costGreen >> costPurple;

        // cout <<"Enter No. of participants: ";
        cin >> p;

        // initialize arrays for holding participants data
        vector<int> greenBal, purpleBal;

        // obtain data for each participant
        for (int i = 0; i < p; i++) {

            // cout <<"Case #"<< i <<" Enter participants result (0 or 1): ";
            int resultGreen, resultPurple;
            cin >> resultGreen >> resultPurple;

            greenBal.push_back(resultGreen);
            purpleBal.push_back(resultPurple);
        }

        int greenCount = 0, purpleCount = 0;

        for (int k = 0; k < p; k++) {
            if (greenBal[k]) greenCount++;
            if (purpleBal[k]) purpleCount++;
        }

        // Calculate the sums
        int sum1 = greenCount*costGreen + purpleCount*costPurple;
        int sum2 = greenCount*costPurple + purpleCount*costGreen;

        // Determine the minimum cost
        cout <<  min(sum1, sum2) << "\n";
    }

    return 0;
}